# GET ALL PARAMETERS USABLE FOR CUSTOM CONFIG BY RUNNING:
## --------------------------------------------------------------
## |          rosrun mrs_uav_core get_public_params.py          #
## --------------------------------------------------------------

mrs_uav_managers:

  estimation_manager:

    # loaded state estimator plugins
    state_estimators: [
      "passthrough",
    ]

    initial_state_estimator: "passthrough" # will be used as the first state estimator
    agl_height_estimator: "" # only slightly filtered height for checking min height (not used in control feedback)

  uav_manager:

    takeoff:

      during_takeoff:
        controller: "MpcController"
        tracker: "LandoffTracker"

      after_takeoff:
        controller: "Se3Controller"
        tracker: "MpcTracker"

    midair_activation:

      during_activation:
        controller: "MidairActivationController"
        tracker: "MidairActivationTracker"

      after_activation:
        controller: "Se3Controller"
        tracker: "MpcTracker"

mrs_uav_trackers:

  mpc_tracker:

    # mrs collision avoidance
    collision_avoidance:

      enabled:           false # disabling this will stop this UAV to react to others, but it will still transmit data to others
      enabled_passively: false # disabling this will stop this uav to even transmit its data to others (only if enabled: false)
